test/common.py

```diff
@@ -661,6 +661,10 @@ def setUp(self):
           # expect this.  --no-clean can be used to keep the old contents for the new test
           # run. This can be useful when iterating on a given test with extra files you want to keep
           # around in the output directory.
+          #
+          # Note that we must make the directory writable, as some tests might
+          # change permissions.
+          make_dir_writeable(self.working_dir)
           utils.delete_contents(self.working_dir)
```

Comment:
Can you create a `force_delete_contents()` helper from these two lines and then call it from here, and also from `self.clear()`?  I think `self.clear()` should do this too, no?

Reply:
I'm not sure this is fully general. In particular it doesn't adjust permissions of internal files and subdirectories. Do you think it's still worth making a helper out of it?

Reply:
What I mean is that `self.clear()` should be as powerful and cleaning up as this code here.  I think it they should use the same code for their cleanup because they are trying to do exactly the same thing.

Reply:
Also I'm not proposing a helper in the emscripten code base.. just a helper here in the test/common.py file just lik the existing `force_delete_dir` helper that lives here just above.

Reply:
In fact it seems to perfectly mirror the existing `force_delete_dir`

Reply:
> In particular it doesn't adjust permissions of internal files and subdirectories

I'm not sure what you mean by this?  `make_dir_writeable` does work recursively IIUC

Reply:
Ah, that's what I was missing, thanks. Now I see what you mean, and that all makes sense. Done.